OAIMetadataFormat_JATS.php

```diff
@@ -18,525 +18,611 @@
-			'</u>' 	=> '</underline>',
-		];
-
-		return str_replace(array_keys($mappings), array_values($mappings), $htmlTitle);
-	}
+                break;
+            }
+        }
+
+        // Issue ID
+        $match = $xpath->query("//article/front/article-meta/issue-id");
+        if ($match->length) {
+            $match->item(0)->appendChild($doc->createTextNode($issue->getId()));
+        } else {
+            $issueIdNode = $this->_addChildInOrder($articleMetaNode, $doc->createElement('issue-id'));
+            $issueIdNode->appendChild($doc->createTextNode($issue->getId()));
+        }
+
+        // Issue cover page
+        if ($coverUrl = $issue->getLocalizedCoverImageUrl()) {
+            $customMetaGroupNode = $this->_addChildInOrder($articleMetaNode, $doc->createElement('custom-meta-group'));
+            $customMetaNode = $customMetaGroupNode->appendChild($doc->createElement('custom-meta'));
+            $metaNameNode = $customMetaNode->appendChild($doc->createElement('meta-name'));
+            $metaNameNode->appendChild($doc->createTextNode('issue-cover'));
+            $metaValueNode = $customMetaNode->appendChild($doc->createElement('meta-value'));
+            $inlineGraphicNode = $metaValueNode->appendChild($doc->createElement('inline-graphic'));
+            $inlineGraphicNode->setAttribute('xmlns:xlink', 'http://www.w3.org/1999/xlink');
+            $inlineGraphicNode->setAttribute('xlink:href', $coverUrl);
+        }
+
+        // Article type
+        if ($articleType = strtolower(trim($section->getLocalizedIdentifyType()))) {
+            $articleNode = $xpath->query("//article")->item(0);
+            $articleNode->setAttribute('article-type', $articleType);
+        }
+
+        // Editorial team
+        $userGroups = UserGroup::withContextIds([$journal->getId()])->get();
+        $journalMetaNode = $xpath->query('//article/front/journal-meta')->item(0);
+        $contribGroupNode = $this->_addChildInOrder($journalMetaNode, $doc->createElement('contrib-group'));
+        $keyContribTypeMapping = [
+            'default.groups.name.manager' => 'jmanager',
+            'default.groups.name.editor' => 'editor',
+            'default.groups.name.sectionEditor' => 'secteditor',
+        ];
+        foreach ($userGroups as $userGroup) {
+            if (!isset($keyContribTypeMapping[$userGroup->nameLocaleKey])) {
+                continue;
+            }
+
+            $users = Repo::user()->getCollector()->filterByUserGroupIds([$userGroup->id])->getMany();
+            foreach ($users as $user) {
+                $contribNode = $contribGroupNode->appendChild($doc->createElement('contrib'));
+                $contribNode->setAttribute('contrib-type', $keyContribTypeMapping[$userGroup->nameLocaleKey]);
+                $nameNode = $contribNode->appendChild($doc->createElement('name'));
+                $surname = method_exists($user, 'getLastName') ? $user->getLastName() : $user->getLocalizedFamilyName();
```

Comment:
Use `getFamilyName` (`getLastName` is long gone), and don't use `getLocalized`...; instead, since users are guaranteed to have required data in the site's primary locale, get the family name using that. (Also below for the given name.)

It's probably worth looking over JATS to see whether we can better align OJS metadata into it -- e.g. JATS would probably accept the optional full name field. This would need to be lined up against the JATS Template plugin, for consistency's sake.

Reply:
I see that in the JATS template plugin we are already using the preferred public name for authors in a `name-alternatives` tag, so we could use the same implementation here for editorial team members - if that sounds good, then I can go ahead and make that change and do a scan for anything else in the JATS template plugin that could be carried over here.

Reply:
Yes please!